src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs

```diff
@@ -29,99 +29,31 @@ internal sealed partial class SerializableOptionSet : OptionSet
         /// </summary>
         private readonly WorkspaceOptionSet _workspaceOptionSet;
 
-        /// <summary>
-        /// Prefetched option values applicable for <see cref="_languages"/>.
-        /// </summary>
-        private readonly ImmutableDictionary<OptionKey, object?> _serializableOptionValues;
-
-        /// <summary>
-        /// Set of changed options in this option set which are serializable.
-        /// </summary>
-        private readonly ImmutableHashSet<OptionKey> _changedOptionKeysSerializable;
-
         /// <summary>
         /// Set of changed options in this option set which are non-serializable.
         /// </summary>
         private readonly ImmutableHashSet<OptionKey> _changedOptionKeysNonSerializable;
 
-        /// <summary>
-        /// Set of languages referenced in <see cref="_serializableOptionValues"/>.  Cached
-        /// only so we can shortcircuit <see cref="UnionWithLanguages"/>.
-        /// </summary>
-        private readonly Lazy<ImmutableHashSet<string>> _languages;
-
         private SerializableOptionSet(
             WorkspaceOptionSet workspaceOptionSet,
-            ImmutableDictionary<OptionKey, object?> values,
-            ImmutableHashSet<OptionKey> changedOptionKeysSerializable,
             ImmutableHashSet<OptionKey> changedOptionKeysNonSerializable)
```

Comment:
what's the current use case for changedOptionKeysNonSerializable?  how does that work? (feel free to not have an answer... i'm just still uncertain about those guys).